site/content/blog/2019-12-18-cloud-monitoring.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Monitoring a Multi-Cluster Environment Using Prometheus Federation and Grafana"
+slug: cloud-monitoring
+date: 2019-12-18T00:00:00-04:00
+author: Stylianos Rigas
+github: stylianosrigas
+community: stylianos.rigas
+---
+
+Monitoring the state of your clusters is an effective way to discover bottlenecks in your multi-cluster production environment. It is one of the key challenges that development teams are facing and factors such as the team experience as well as the number and distribution of the applications can make things even more complex.
+
+Better monitoring can help identify single points of failure and being able to get information about things like the number of applications running on each node or the CPU and memory performance can help to get closer to zero downtime deployments.
+
+At Mattermost Cloud we run multiple Kubernetes clusters in AWS, deployed with our own Mattermost provisioning tool. The clusters are deployed across multiple AWS VPCs and this cluster distribution led to some challenges which are highlighted below:
+
+- Monitoring tooling
+- Data transfer cost
+- DNS resolution
+
+The aim of this blog is to go through each one of the challenges we faced and try to explain the path we followed to reach the end solution.
+
+#### Monitoring Tooling
+
+For the monitoring tooling a combination of the open source Prometheus and Grafana tools was selected. Prometheus acts at the storage and Grafana as the interface for visualization of the monitoring data via dashboards and panels.
+
+Grafana offers an easy to read interface that can be easily deployed and managed in a k8s cluster. By supporting multiple datasources (Prometheus, MySQL, AWS CloudWatch, etc.), it can be used to monitor the whole infrastructure.
```

Comment:
```suggestion
Grafana offers an easy to read interface that can be easily deployed and managed in a Kubernetes cluster. By supporting multiple datasources (Prometheus, MySQL, AWS CloudWatch, etc.), it can be used to monitor the whole infrastructure.
```